components/gitpod-db/src/typeorm/migration/1662040283793-AddUsageTable.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class AddUsageTable1662040283793 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(
+            `CREATE TABLE \`d_b_usage\` (
+                \`id\` char(36) NOT NULL,
+                \`attributionId\` varchar(255) NOT NULL,
+                \`description\` varchar(255) NOT NULL,
+                \`creditCents\` bigint NOT NULL,
+                \`effectiveTime\` varchar(255) NOT NULL,
+                \`creationTime\` varchar(255) NOT NULL,
+                \`kind\` varchar(255) NOT NULL,
+                \`workspaceInstanceId\` char(36) NULL,
+                \`draft\` BOOLEAN NOT NULL,
+                \`metadata\` text NULL,
+                \`_lastModified\` timestamp(6) NOT NULL,
```

Comment:
This should have an auto-update property on it

Reply:
It has implicitly. Should I make it explicit?

Reply:
will make it explicit

Reply:
I think ensuring that the DB does this for us guards against mistakes we make in code and forget to update the timestamp. It would also reduce the amount of code (the setting of the last modified).

I'd personally like that the DB (explicit) takes care of this rather than doing it our application layer.

Reply:
Yes, I meant it does it for us, because timestamp(6) NOT NULL does insert/update the current timestamp. But writing it explicitly into the migration file is better.

Reply:
What I'm after is this:
```
 ts TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
```
Is that the behavuour the current statement would have?

Reply:
yes. @andrew-farries was sharing [this](https://stackoverflow.com/questions/22860351/mysql-column-type-timestamp-implicitly-includes-not-null-default-current-time) the other day.
But it's better to be explicit.